File: src/2025-03-04-12-20-21--deepseek-vr-0304/go/exercises/practice/bottle-song/bottle_song.go
package bottlesong

import (
	"fmt"
	"strconv"
	"strings"
)

func Recite(startBottles, takeDown int) []string {
	numberWords := map[int]string{
		0:  "no",
		1:  "One",
		2:  "Two",
		3:  "Three",
		4:  "Four",
		5:  "Five",
		6:  "Six",
		7:  "Seven",
		8:  "Eight",
		9:  "Nine",
		10: "Ten",
	}

	getWord := func(n int) string {
		if word, ok := numberWords[n]; ok {
			return word
		}
		return strconv.Itoa(n)
	}

	var verses []string

	for i := 0; i < takeDown; i++ {
		current := startBottles - i
		currentWord := getWord(current)
		currentBottles := "bottles"
		if current == 1 {
			currentBottles = "bottle"
		}

		line1 := fmt.Sprintf("%s green %s hanging on the wall,", currentWord, currentBottles)
		line2 := line1
		line3 := "And if one green bottle should accidentally fall,"

		next := current - 1
		nextWord := strings.ToLower(getWord(next))

		nextBottles := "bottles"
		if next == 1 {
			nextBottles = "bottle"
		}

		line4 := fmt.Sprintf("There'll be %s green %s hanging on the wall.", nextWord, nextBottles)

		verses = append(verses, line1, line2, line3, line4)

		if i != takeDown-1 {
			verses = append(verses, "")
		}
	}

	return verses
}

File: src/2025-03-04-12-20-21--deepseek-vr-0304/javascript/exercises/practice/bottle-song/bottle-song.js
export const recite = (initialBottlesCount, takeDownCount) => {
  const numberWords = ['No', 'One', 'Two', 'Three', 'Four', 'Five', 'Six', 'Seven', 'Eight', 'Nine', 'Ten'];

  const numberToWord = (n) => numberWords[n];

  const pluralize = (n) => n === 1 ? 'bottle' : 'bottles';

  const generateVerse = (current) => {
    const nextN = current - 1;
    const currentWord = numberToWord(current);
    const nextWord = numberToWord(nextN);
    const currentBottles = `${currentWord} green ${pluralize(current)}`;
    const nextBottles = nextN === 0 
      ? 'no green bottles' 
      : `${nextWord.toLowerCase()} green ${pluralize(nextN)}`;
    
    const lines = [
      `${currentBottles} hanging on the wall,`,
      `${currentBottles} hanging on the wall,`,
      `And if one green bottle should accidentally fall,`,
      `There'll be ${nextBottles} hanging on the wall.`
    ];
    
    return lines.join('\n');
  };

  const verses = [];
  for (let i = 0; i < takeDownCount; i++) {
    const current = initialBottlesCount - i;
    if (current < 1) break;
    verses.push(generateVerse(current));
  }

  return verses.join('\n\n');
};
